crates/ecs/src/lib.rs

```diff
@@ -460,8 +464,9 @@ pub type ArchetypeResult<T, E = ArchetypeError> = Result<T, E>;
 /// Stores components associated with entity ids.
 #[derive(Debug, Default)]
 struct Archetype {
-    component_typeid_to_component_vec: HashMap<TypeId, Box<dyn ComponentVec>>,
-    entity_to_component_index: HashMap<Entity, ComponentIndex>,
+    component_typeid_to_component_vec: FnvHashMap<TypeId, Box<dyn ComponentVec>>,
+    entity_to_component_index:
+        HashMap<Entity, ComponentIndex, BuildHasherDefault<NoHashHasher<Entity>>>,
```

Comment:
Could you type alias this `HashMap` into something like `NoHashHashMap<K, V>` as well?